ndk-build/src/manifest.rs

```diff
@@ -1,258 +1,322 @@
+    #[serde(default)]
+    pub label: String,
+
+    #[serde(rename(serialize = "meta-data"))]
+    #[serde(default)]
+    pub meta_data: Vec<MetaData>,
+    #[serde(default)]
+    pub activity: Activity,
+}
+
+/// See https://developer.android.com/guide/topics/manifest/activity-element
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub struct Activity {
+    #[serde(rename(serialize = "android:configChanges"))]
+    #[serde(default = "default_config_changes")]
+    pub config_changes: Option<String>,
+    #[serde(rename(serialize = "android:label"))]
+    pub label: Option<String>,
+    #[serde(rename(serialize = "android:launchMode"))]
+    pub launch_mode: Option<String>,
+    #[serde(rename(serialize = "android:name"))]
+    #[serde(default = "default_activity_name")]
     pub name: String,
-    pub required: bool,
+    #[serde(rename(serialize = "android:screenOrientation"))]
+    pub orientation: Option<String>,
+
+    #[serde(rename(serialize = "meta-data"))]
+    #[serde(default)]
+    pub meta_data: Vec<MetaData>,
+    /// If no `MAIN` action exists in any intent filter, a default `MAIN` filter is serialized.
+    #[serde(serialize_with = "serialize_intents")]
+    #[serde(rename(serialize = "intent-filter"))]
+    #[serde(default)]
+    pub intent_filter: Vec<IntentFilter>,
 }
 
-impl Feature {
-    pub fn to_string(&self) -> String {
-        format!(
-            r#"<uses-feature android:name="{}" android:required="{}"/>"#,
-            &self.name, self.required,
-        )
+impl Default for Activity {
+    fn default() -> Self {
+        Self {
+            config_changes: default_config_changes(),
+            label: None,
+            launch_mode: None,
+            name: default_activity_name(),
+            orientation: None,
+            meta_data: Default::default(),
+            intent_filter: Default::default(),
+        }
     }
 }
 
-#[derive(Debug)]
-pub struct ApplicationMetadata {
-    pub name: String,
-    pub value: String,
-}
+fn serialize_intents<S>(
+    intent_filters: &Vec<IntentFilter>,
+    serializer: S,
+) -> Result<S::Ok, S::Error>
+where
+    S: Serializer,
+{
+    use serde::ser::SerializeSeq;
+
+    let mut seq = serializer.serialize_seq(None)?;
+    for intent_filter in intent_filters {
+        seq.serialize_element(intent_filter)?;
+    }
 
-impl ApplicationMetadata {
-    pub fn to_string(&self) -> String {
-        format!(
-            r#"<meta-data android:name="{}" android:value="{}"/>"#,
-            self.name, self.value
-        )
+    // Check if `intent_filters` contains a `MAIN` action. If not, add a default filter.
+    if intent_filters
+        .iter()
+        .all(|i| i.actions.iter().all(|f| f != "android.intent.action.MAIN"))
+    {
+        seq.serialize_element(&IntentFilter {
+            actions: vec!["android.intent.action.MAIN".to_string()],
+            categories: vec!["android.intent.category.LAUNCHER".to_string()],
+            data: Vec::new(),
```

Comment:
Idr if `vec![]` or `Default::default` is preferred here. Either way, make it consistent with all the other `Vec<>` initializations in this file. Perhaps `..Default::default()` makes more sense here, though being explicit about the last member is probably the better way to go.